docs/api/operators.md

```diff
@@ -0,0 +1,248 @@
+---
+title: Operators
+order: 1
+---
+
+Operators in Wonka allow you to transform values from a source before they are sent to a sink. Wonka has the following operators.
+
+## combine
+
+`combine` two sources together to a single source. The emitted value will be a combination of the two sources, with all values from the first source being emitted with the first value of the second source _before_ values of the second source are emitted.
+
+```reason
+open Wonka;
+open Wonka_sources;
+open Wonka_operators;
+open Wonka_sinks;
+
+let sourceOne = fromArray([|1, 2, 3|]);
+let sourceTwo = fromArray([|4, 5, 6|]);
+
+combine(sourceOne, sourceTwo)
+  |> subscribe((. (_valOne, _valTwo)) => print_int(_valOne + _valTwo));
+
+/* Prints 56789 (1+4, 2+4, 3+4, 3+5, 3+6) to the console. */
+```
+
+```typescript
+import { fromArray, pipe, combine, subscribe } from 'wonka';
+
+const sourceOne = fromArray([1, 2, 3]);
+const sourceTwo = fromArray([4, 5, 6]);
+
+pipe(
+  combine(sourceOne, sourceTwo),
+  subscribe(([valOne, valTwo]) => {
+    console.log(valOne + valTwo);
+  })
+);
+
+// Prints 56789 (1+4, 2+4, 3+4, 3+5, 3+6) to the console.
+```
+
+## concat
+
+`concat` will combine two sources together, subscribing to the next source after the previous source completes.
```

Comment:
One thing that confused me a bit when writing this section – what is the meaningful difference between `concat` and `merge`? They seem to behave the same in very simple cases that I used, but wondering if there are more complex cases that might highlight their differences better. Also, should we include docs on `concatMap` and `concatAll`? I don't fully understand how those work, so it may be better for you to expand there @kitten.

Reply:
We should probably include docs on those, yes, since they’re very useful individually.

Also thinking about how we can add some visualisations somehow.

concat and merge should behave the exact same for synchronous sources due to the nature of pull streams. In an asynchronous push stream scenario they’re different however.

So while merge also goes through sources and pulls from them in order (which I thought was an important detail for predictability) for asynchronous sources merge will just let all values through as they arrive, since it subscribed to them all at the same time (but still in order)

Concat however will also subscribe in order, which means that for asynchronous sources the values remain in order of the sources.

You can even see this difference in code quite easily if you know what to look for. The implementations are very similar with the “applyInnerSource” function subscribing to sources and forwarding their values.

The difference lies in when this function is called. concatMap will only call it once the last source has completed, while mergeMap will eagerly go on and call it immediately for each incoming source.

This further underlines why they behave the same for synchronous sources. This function also pulls values like the sinks do. So for synchronous sources the function is kept busy and keeps running until the source completes, as values immediately come in synchronously.

Reply:
Beautiful description, this makes a lot of sense. I think it'd be great for us to include some simple async examples to make this distinction clear. I'll work on some this weekend and we can continue to refine this further. Thanks again Phil ❤️